### ISSUE TITLE
Volume resize can fail with `freenas-iscsi` driver with a custom iscsi nameTemplate

### ISSUE THREAD

**Issue by d-uzlov** [democratic-csi/democratic-csi#295]
Steps to reproduce:
1. Create an iscsi provisioner with `driver.config.iscsi.nameTemplate` set to `"{{ parameters.[csi.storage.k8s.io/pvc/namespace] }}.{{ parameters.[csi.storage.k8s.io/pvc/name] }}"` and/or `driver.config.iscsi.namePrefix` set to `csi.` (notice `.` at the end)
2. Create a PVC with `100Mi` size
3. Update PVC size to `200Mi`

On the truenas side volume size is changed successfully, but in k8s PVC size does not change.

Relevant log from `describe pvc` command:

`Warning  VolumeResizeFailed  5s (x5 over 11s)  external-resizer org.democratic-csi.truenas-iscsi  resize volume "pvc-21037f21-5d0a-4a56-aed8-dc5d60f8d888" by resizer "org.democratic-csi.truenas-iscsi" failed: rpc error: code = Unknown desc = error reloading iscsi daemon: {"stderr":"bash: line 1: /sys/kernel/scst_tgt/devices/csi.default.test-iscsi/resync_size: No such file or directory\n","code":1}`

Seems like this file doesn't exist: `/sys/kernel/scst_tgt/devices/csi.default.test-iscsi/resync_size`

When I check my truenas instance, I see that the file is indeed named differently:
```bash
$ ls /sys/kernel/scst_tgt/devices/
csi_default_test-iscsi
```

`.` symbols from the name template got replaced with `_` symbols.

democratic-csi version:
`docker.io/democraticcsi/democratic-csi@sha256:b6fd5b5538f444f9d4d202de7728a0b76a56cf7750167f252417713760158ba9`
(this is currently the latest image from the `next` branch)

**Comment by travisghansen** [comment]
Interesting. This looks like it’s on SCALE?

**Comment by d-uzlov** [comment]
Yes, this is on truenas scale.

**Comment by travisghansen** [comment]
Force the controller pod to update to the most recent `next` image and give it a try.

https://github.com/democratic-csi/democratic-csi/commit/a3df4bcca0ee321bbdfbe42ef2ef9adcb9b96aa4

**Comment by d-uzlov** [comment]
The naming issue seems to be solved.
However, now I get an error about permissions.
Logs from csi-driver:
```log
{"host":"truenas-iscsi-democratic-csi-controller-67485d8ff-j6sms","level":"verbose","message":"whoami command: whoami","service":"democratic-csi","timestamp":"2023-05-10T14:40:45.497Z"}
{"host":"truenas-iscsi-democratic-csi-controller-67485d8ff-j6sms","level":"verbose","message":"FreeNAS reloading iscsi daemon: /usr/bin/sudo sh -c echo 1 > /sys/kernel/scst_tgt/devices/csi_default_test-iscsi/resync_size","service":"democratic-csi","timestamp":"2023-05-10T14:40:45.502Z"}
{"host":"truenas-iscsi-democratic-csi-controller-67485d8ff-j6sms","level":"error","message":"handler error - driver: FreeNASSshDriver method: ControllerExpandVolume error: {\"name\":\"GrpcError\",\"code\":2,\"message\":\"error reloading iscsi daemon: {\\\"stderr\\\":\\\"bash: line 1: /sys/kernel/scst_tgt/devices/csi_default_test-iscsi/resync_size: Permission denied\\\\n\\\",\\\"code\\\":1}\"}","service":"democratic-csi","timestamp":"2023-05-10T14:40:45.507Z"}
```
I can guess that `/usr/bin/sudo sh -c echo 1 > /sys/kernel/scst_tgt/devices/csi_default_test-iscsi/resync_size` doesn't actually use sudo for writing the file, because `> /path/to/file` is executed in the outer shell, before calling sudo. Though, I didn't dig the code  to check if I'm correct about how the commands are executed.

**Comment by isugimpy** [comment]
Would it be possible to just change this from doing a redirection to the file to instead just `echo 1 | sudo tee /sys/kernel/scst_tgt/devices/csi_default_test-iscsi/resync_size`?  That should alleviate the issue.  Selfishly, I'd love to see this fixed as it just bit me a few minutes ago.

**Comment by sfxworks** [comment]
oof just ran into this myself

```
  Normal   Resizing            17s (x2 over 21s)  external-resizer org.democratic-csi.iscsi  External resizer is resizing volume pvc-3e2c9064-c12e-4fb9-856a-13e610fb5f1d
  Warning  VolumeResizeFailed  17s                external-resizer org.democratic-csi.iscsi  resize volume "pvc-3e2c9064-c12e-4fb9-856a-13e610fb5f1d" by resizer "org.democratic-csi.iscsi" failed: rpc error: code = Unknown desc = error reloading iscsi daemon: {"stderr":"bash: line 1: /sys/kernel/scst_tgt/devices/csi-pvc-3e2c9064-c12e-4fb9-856a-13e610fb5f1d-cluster/resync_size: Permission denied\n","code":1}
```

**Comment by travisghansen** [comment]
I just committed a fix for this I would like you to try. If you could deploy with the `next` image that would be great.

https://github.com/democratic-csi/democratic-csi/commit/57f74ce665ada752430d3a5721aaffaca635f3a1

**Comment by travisghansen** [comment]
Fix should be released in `v1.9.0`. Reopen if you see the behavior continue after upgrading.
